src/Models/Entry.php

```diff
@@ -74,16 +74,69 @@ public function __construct(
 
     /**
      * List all diary entries for the current user
+     * @param array List of fields to be filtered by
      * @return array(Entry) List of diary entries
      */
-    public static function getEntriesForCurrentUser() {
+    public static function getEntriesForCurrentUser($filter_by = array()) {
         if (Session::getInstance()->getUser() == null) return array();
 
+        $filter_by_condition = "";
+        if (count($filter_by) > 0) {
+            /*
+             * Only allow certain fields to be filtered
+             */ 
+            $filter_by = array_filter($filter_by, function($key) {
+                return $key == 'category_id' || $key == 'publish_date';
+            }, ARRAY_FILTER_USE_KEY);
```

Comment:
Isn't this already filtered in the page controller by only copying certain values to `$filter_by`?

Reply:
True, well indirectly on HomePage Controller, it only copies category_id and publish_date into filter_by array. On the Entry function, it checks if only those are set.

What do you think, should we have this double check or remove one?

Reply:
I think it would be safe to remove the second filter, but I guess it doesn't really matter, since the function can't process other keys anyways :)

Reply:
Let's leave it as it is, in case we want to expand the filter in the future.